### PR TITLE
Bump clap from 3.2.25 to 4.4.12 and MSRV from 1.60.0 to 1.70.0

### PR DESCRIPTION
**Pull request by rustworthy**
Move from clap@3 to clap@4 and set msrv to 1.70

- - -
This change is [Reviewable](https://reviewable.io/reviews/jonhoo/faktory-rs/34)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -31,7 +31,8 @@ url = "2"
 atomic-option = "0.1"
 fnv = "1.0.5"
 native-tls = { version = "0.2", optional = true }
-clap = { version = "3.1.0", optional = true }
+openssl = { version = "0.10.61", optional = true }
```
Comment (jonhoo) [inline Cargo.toml:34]: Why do we now need a direct dependency on `openssl`?

File: .github/workflows/check.yml [hunk 2]
```diff
@@ -89,7 +89,7 @@ jobs:
     # https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability
     strategy:
       matrix:
-        msrv: [1.57.0]
+        msrv: [1.70.0]
```
Comment (jonhoo) [inline .github/workflows/check.yml:92]: Why this large MSRV bump? That's pretty significant (and recent). Is it required by anything we depend on?

File: Cargo.toml (hunk 1 above)
Comment (rustworthy) [reply Cargo.toml:34]: direct dependency with version pinned will prevent the reset of openssl to 0.9.xx in 'minimal versions job' which will error.  From the latest workflow [run](https://github.com/jonhoo/faktory-rs/actions/runs/7237492281/job/19717219623):
image

If we are ok with that job failing, this direct dependency is redundant. 

There is also probably a better way out, which I am missing..

File: .github/workflows/check.yml (hunk 2 above)
Comment (rustworthy) [reply .github/workflows/check.yml:92]: It is current [clap](https://github.com/clap-rs/clap/blob/master/Cargo.toml)'s msrv.

As far as I understand they have been forced to bump it due to [anstyle](https://github.com/rust-cli/anstyle).

image

File: Cargo.toml (hunk 1 above)
Comment (jonhoo) [reply Cargo.toml:34]: This should now be fixed by https://github.com/jonhoo/faktory-rs/pull/40

**Comment by jonhoo** [comment]
I think many of the changes in this can now go away since the various CI fixes I've landed on `main`.

As for the clap MSRV bump, that's unfortunate, but I guess we'll take it. After all, people _can_ still install with an older MSRV, they just need [an MSRV-aware resolve](https://github.com/rust-lang/cargo/issues/9930).

**Comment by jonhoo** [comment]
I wanted to adjust the clap code slightly, so ended up committing this directly in https://github.com/jonhoo/faktory-rs/commit/2c728b16a600124167956e56751a68475b96114b so we didn't need to go back and forth another time :)

**Comment by jonhoo** [comment]
Released in 0.12.2 :tada:

**Pull request closed without merging**